### PR TITLE
Update icalendar tool to python3

### PR DESCRIPTION
**Pull request by jacadzaca**
This pull request attempts to solve issue #316 . It also:

1. breaks backwards compatibility - I've removed the -o option, the program now always prints the output to the stdout. While introducing clutter, the -o option dose not introduce much value, because the user can just pipe the output. 
2. allows the program to take multiple calendar files as input.
3. changes the output formatting
4. makes the program take a calendar file directly as an input (e.g 'icalendar view test.ics' => 'icalendar test.ics'
5. makes the program output a 'Duration' row

If it's not okay for me to introduce any of these changes, I will happily revert any of them :)

What the new output looks like:
new_output

### CODE REVIEW

**Comment by jacadzaca** [comment]
@jenkins-plone-org please run jobs

**Comment by Thorbenl** [comment]
whats the status on this? :D Kind of a road blocker for us that it doesnt work on 3.x...

**Comment by niccokunzmann** [comment]
@jacadzaca, we are in the process of setting up maintenance for this repository again, see https://github.com/collective/icalendar/discussions/360. It will take some time but I hope, we will get further towards integrating your changes. Thanks a lot for the effort and patience already!
If you like to run the latest tests for this PR, you can rebase/merge master.

Personally, I wonder about two things:
- in Windows, I would like to have the -o option
- can there be tests for that? Because that would prevent people in the future to break your code...

**Comment by niccokunzmann** [comment]
Uh ... tests running ...

**Comment by niccokunzmann** [comment]
Cool. Only the test fails because the output differs.
Maybe the output is colored?
https://github.com/collective/icalendar/runs/7940793431?check_suite_focus=true#step:6:46

**Comment by jacadzaca** [comment]
When I run the test locally it seems to pass, am I invoking the test improperly or something?
icalendar_test_000
icalendar_test2_000

**Comment by niccokunzmann** [comment]
I can see this:

> 10:34:00 2022\n    End        : Sat Aug 2[592 chars]\n\n' != '    [192 chars]g 20 08:34:00 2022

Maybe, you want to print that... Let's see if I can reproduce it...

**Comment by niccokunzmann** [comment]
I am on Ubuntu and I see this:

```
icalendar$ git pull https://github.com/jacadzaca/icalendar.git fix-icalendar-tool
remote: Enumerating objects: 50, done.
remote: Counting objects: 100% (50/50), done.
remote: Compressing objects: 100% (13/13), done.
remote: Total 50 (delta 29), reused 50 (delta 29), pack-reused 0
Unpacking objects: 100% (50/50), 12.92 KiB | 18.00 KiB/s, done.
From https://github.com/jacadzaca/icalendar
 * branch            fix-icalendar-tool -> FETCH_HEAD
Auto-merging CHANGES.rst
Auto-merging docs/credits.rst
Merge made by the 'ort' strategy.
 CHANGES.rst                          |   7 ++
 docs/credits.rst                     |   1 +
 src/icalendar/cli.py                 | 133 ++++++++++++++---------------------
 src/icalendar/tests/test_cli_tool.py |  72 +++++++++++++++++++
 4 files changed, 133 insertions(+), 80 deletions(-)
 mode change 100644 => 100755 src/icalendar/cli.py
 create mode 100644 src/icalendar/tests/test_cli_tool.py
sabin@schneidebrett:~/icalendar$ tox
/home/sabin/icalendar/.tox/py38/lib/python3.8/site-packages/setuptools/config/setupcfg.py:463: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
  warnings.warn(msg, warning_class)
py38 develop-inst-nodeps: /home/sabin/icalendar
py38 installed: attrs==22.1.0,backports.zoneinfo==0.2.1,coverage==6.4.4,-e git+ssh://git@github.com/niccokunzmann/icalendar@de54b76abb13bfac42b07050dc9fac42f514a6fa#egg=icalendar,iniconfig==1.1.1,packaging==21.3,pluggy==1.0.0,py==1.11.0,pyparsing==3.0.9,pytest==7.1.2,python-dateutil==2.8.2,pytz==2022.2.1,six==1.16.0,tomli==2.0.1
py38 run-test-pre: PYTHONHASHSEED='3163265074'
py38 run-test: commands[0] | coverage run --source=src/icalendar '--omit=*/tests/*' --module pytest
============================= test session starts ==============================
platform linux -- Python 3.8.13, pytest-7.1.2, pluggy-1.0.0
cachedir: .tox/py38/.pytest_cache
rootdir: /home/sabin/icalendar, configfile: setup.cfg
collected 100 items                                                            

src/icalendar/tests/test_cli_tool.py F                                   [  1%]
src/icalendar/tests/test_encoding.py ....                                [  5%]
src/icalendar/tests/test_fixed_issues.py .....................           [ 26%]
src/icalendar/tests/test_icalendar.py .........                          [ 35%]
src/icalendar/tests/test_multiple.py .                                   [ 36%]
src/icalendar/tests/test_property_params.py .......                      [ 43%]
src/icalendar/tests/test_recurrence.py ..                                [ 45%]
src/icalendar/tests/test_time.py ..                                      [ 47%]
src/icalendar/tests/test_timezoned.py ..........                         [ 57%]
src/icalendar/tests/test_unit_cal.py ...........                         [ 68%]
src/icalendar/tests/test_unit_caselessdict.py ....                       [ 72%]
src/icalendar/tests/test_unit_parser_tools.py ..                         [ 74%]
src/icalendar/tests/test_unit_prop.py .........................          [ 99%]
src/icalendar/tests/test_unit_tools.py .                                 [100%]

=================================== FAILURES ===================================
________________________ CLITool.test_output_is_proper _________________________

self = <icalendar.tests.test_cli_tool.CLITool testMethod=test_output_is_proper>

    def test_output_is_proper(self):
        calendar = Calendar.from_ical(INPUT)
        output = ''
        for event in calendar.walk('vevent'):
            output += cli.view(event) + '\n\n'
>       self.assertEqual(PROPER_OUTPUT, output)
E       AssertionError: '    [192 chars]g 20 10:34:00 2022\n    End        : Sat Aug 2[592 chars]\n\n' != '    [192 chars]g 20 09:34:00 2022\n    End        : Sat Aug 2[592 chars]\n\n'
E       Diff is 1305 characters long. Set self.maxDiff to None to see it.

src/icalendar/tests/test_cli_tool.py:68: AssertionError
=========================== short test summary info ============================
FAILED src/icalendar/tests/test_cli_tool.py::CLITool::test_output_is_proper
========================= 1 failed, 99 passed in 5.30s =========================
ERROR: InvocationError for command /home/sabin/icalendar/.tox/py38/bin/coverage run --source=src/icalendar '--omit=*/tests/*' --module pytest (exited with code 1)
/home/sabin/icalendar/.tox/py39/lib/pypy3.9/site-packages/setuptools/config/setupcfg.py:463: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
  warnings.warn(msg, warning_class)
py39 develop-inst-noop: /home/sabin/icalendar
py39 installed: attrs==22.1.0,cffi==1.15.0,coverage==6.4.3,greenlet==0.4.13,hpy==0.0.3,-e git+ssh://git@github.com/niccokunzmann/icalendar@de54b76abb13bfac42b07050dc9fac42f514a6fa#egg=icalendar,iniconfig==1.1.1,packaging==21.3,pluggy==1.0.0,py==1.11.0,pyparsing==3.0.9,pytest==7.1.2,python-dateutil==2.8.2,pytz==2022.2.1,readline==6.2.4.1,six==1.16.0,tomli==2.0.1
py39 run-test-pre: PYTHONHASHSEED='3163265074'
py39 run-test: commands[0] | coverage run --source=src/icalendar '--omit=*/tests/*' --module pytest
============================= test session starts ==============================
platform linux -- Python 3.9.12[pypy-7.3.9-final], pytest-7.1.2, pluggy-1.0.0
cachedir: .tox/py39/.pytest_cache
rootdir: /home/sabin/icalendar, configfile: setup.cfg
collected 100 items                                                            

src/icalendar/tests/test_cli_tool.py F                                   [  1%]
src/icalendar/tests/test_encoding.py ....                                [  5%]
src/icalendar/tests/test_fixed_issues.py .....................           [ 26%]
src/icalendar/tests/test_icalendar.py .........                          [ 35%]
src/icalendar/tests/test_multiple.py .                                   [ 36%]
src/icalendar/tests/test_property_params.py .......                      [ 43%]
src/icalendar/tests/test_recurrence.py ..                                [ 45%]
src/icalendar/tests/test_time.py ..                                      [ 47%]
src/icalendar/tests/test_timezoned.py ..........                         [ 57%]
src/icalendar/tests/test_unit_cal.py ...........                         [ 68%]
src/icalendar/tests/test_unit_caselessdict.py ....                       [ 72%]
src/icalendar/tests/test_unit_parser_tools.py ..                         [ 74%]
src/icalendar/tests/test_unit_prop.py .........................          [ 99%]
src/icalendar/tests/test_unit_tools.py .                                 [100%]

=================================== FAILURES ===================================
________________________ CLITool.test_output_is_proper _________________________

self = <icalendar.tests.test_cli_tool.CLITool testMethod=test_output_is_proper>

    def test_output_is_proper(self):
        calendar = Calendar.from_ical(INPUT)
        output = ''
        for event in calendar.walk('vevent'):
            output += cli.view(event) + '\n\n'
>       self.assertEqual(PROPER_OUTPUT, output)
E       AssertionError: '    [192 chars]g 20 10:34:00 2022\n    End        : Sat Aug 2[592 chars]\n\n' != '    [192 chars]g 20 09:34:00 2022\n    End        : Sat Aug 2[592 chars]\n\n'
E       Diff is 1305 characters long. Set self.maxDiff to None to see it.

src/icalendar/tests/test_cli_tool.py:68: AssertionError
=========================== short test summary info ============================
FAILED src/icalendar/tests/test_cli_tool.py::CLITool::test_output_is_proper
======================== 1 failed, 99 passed in 12.22s =========================
ERROR: InvocationError for command /home/sabin/icalendar/.tox/py39/bin/coverage run --source=src/icalendar '--omit=*/tests/*' --module pytest (exited with code 1)
py310 recreate: /home/sabin/icalendar/.tox/py310
py310 installdeps: pytest, coverage
py310 develop-inst: /home/sabin/icalendar
py310 installed: attrs==22.1.0,coverage==6.4.4,-e git+ssh://git@github.com/niccokunzmann/icalendar@de54b76abb13bfac42b07050dc9fac42f514a6fa#egg=icalendar,iniconfig==1.1.1,packaging==21.3,pluggy==1.0.0,py==1.11.0,pyparsing==3.0.9,pytest==7.1.2,python-dateutil==2.8.2,pytz==2022.2.1,six==1.16.0,tomli==2.0.1
py310 run-test-pre: PYTHONHASHSEED='3163265074'
py310 run-test: commands[0] | coverage run --source=src/icalendar '--omit=*/tests/*' --module pytest
============================= test session starts ==============================
platform linux -- Python 3.10.4, pytest-7.1.2, pluggy-1.0.0
cachedir: .tox/py310/.pytest_cache
rootdir: /home/sabin/icalendar, configfile: setup.cfg
collected 100 items                                                            

src/icalendar/tests/test_cli_tool.py F                                   [  1%]
src/icalendar/tests/test_encoding.py ....                                [  5%]
src/icalendar/tests/test_fixed_issues.py .....................           [ 26%]
src/icalendar/tests/test_icalendar.py .........                          [ 35%]
src/icalendar/tests/test_multiple.py .                                   [ 36%]
src/icalendar/tests/test_property_params.py .......                      [ 43%]
src/icalendar/tests/test_recurrence.py ..                                [ 45%]
src/icalendar/tests/test_time.py ..                                      [ 47%]
src/icalendar/tests/test_timezoned.py ..........                         [ 57%]
src/icalendar/tests/test_unit_cal.py ...........                         [ 68%]
src/icalendar/tests/test_unit_caselessdict.py ....                       [ 72%]
src/icalendar/tests/test_unit_parser_tools.py ..                         [ 74%]
src/icalendar/tests/test_unit_prop.py .........................          [ 99%]
src/icalendar/tests/test_unit_tools.py .                                 [100%]

=================================== FAILURES ===================================
________________________ CLITool.test_output_is_proper _________________________

self = <icalendar.tests.test_cli_tool.CLITool testMethod=test_output_is_proper>

    def test_output_is_proper(self):
        calendar = Calendar.from_ical(INPUT)
        output = ''
        for event in calendar.walk('vevent'):
            output += cli.view(event) + '\n\n'
>       self.assertEqual(PROPER_OUTPUT, output)
E       AssertionError: '    [192 chars]g 20 10:34:00 2022\n    End        : Sat Aug 2[592 chars]\n\n' != '    [192 chars]g 20 09:34:00 2022\n    End        : Sat Aug 2[592 chars]\n\n'
E       Diff is 1305 characters long. Set self.maxDiff to None to see it.

src/icalendar/tests/test_cli_tool.py:68: AssertionError
=========================== short test summary info ============================
FAILED src/icalendar/tests/test_cli_tool.py::CLITool::test_output_is_proper
========================= 1 failed, 99 passed in 1.55s =========================
ERROR: InvocationError for command /home/sabin/icalendar/.tox/py310/bin/coverage run --source=src/icalendar '--omit=*/tests/*' --module pytest (exited with code 1)
/home/sabin/icalendar/.tox/pypy3/lib/pypy3.8/site-packages/setuptools/config/setupcfg.py:463: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
  warnings.warn(msg, warning_class)
pypy3 develop-inst-noop: /home/sabin/icalendar
pypy3 installed: attrs==22.1.0,cffi==1.15.0,coverage==6.4.3,greenlet==0.4.13,hpy==0.0.3,-e git+ssh://git@github.com/niccokunzmann/icalendar@de54b76abb13bfac42b07050dc9fac42f514a6fa#egg=icalendar,iniconfig==1.1.1,packaging==21.3,pluggy==1.0.0,py==1.11.0,pyparsing==3.0.9,pytest==7.1.2,python-dateutil==2.8.2,pytz==2022.2.1,readline==6.2.4.1,six==1.16.0,tomli==2.0.1
pypy3 run-test-pre: PYTHONHASHSEED='3163265074'
pypy3 run-test: commands[0] | coverage run --source=src/icalendar '--omit=*/tests/*' --module pytest
============================= test session starts ==============================
platform linux -- Python 3.8.13[pypy-7.3.9-final], pytest-7.1.2, pluggy-1.0.0
cachedir: .tox/pypy3/.pytest_cache
rootdir: /home/sabin/icalendar, configfile: setup.cfg
collected 100 items                                                            

src/icalendar/tests/test_cli_tool.py F                                   [  1%]
src/icalendar/tests/test_encoding.py ....                                [  5%]
src/icalendar/tests/test_fixed_issues.py .....................           [ 26%]
src/icalendar/tests/test_icalendar.py .........                          [ 35%]
src/icalendar/tests/test_multiple.py .                                   [ 36%]
src/icalendar/tests/test_property_params.py .......                      [ 43%]
src/icalendar/tests/test_recurrence.py ..                                [ 45%]
src/icalendar/tests/test_time.py ..                                      [ 47%]
src/icalendar/tests/test_timezoned.py .s.s......                         [ 57%]
src/icalendar/tests/test_unit_cal.py ...........                         [ 68%]
src/icalendar/tests/test_unit_caselessdict.py ....                       [ 72%]
src/icalendar/tests/test_unit_parser_tools.py ..                         [ 74%]
src/icalendar/tests/test_unit_prop.py .........................          [ 99%]
src/icalendar/tests/test_unit_tools.py .                                 [100%]

=================================== FAILURES ===================================
________________________ CLITool.test_output_is_proper _________________________

self = <icalendar.tests.test_cli_tool.CLITool testMethod=test_output_is_proper>

    def test_output_is_proper(self):
        calendar = Calendar.from_ical(INPUT)
        output = ''
        for event in calendar.walk('vevent'):
            output += cli.view(event) + '\n\n'
>       self.assertEqual(PROPER_OUTPUT, output)
E       AssertionError: '    [192 chars]g 20 10:34:00 2022\n    End        : Sat Aug 2[592 chars]\n\n' != '    [192 chars]g 20 09:34:00 2022\n    End        : Sat Aug 2[592 chars]\n\n'
E       Diff is 1305 characters long. Set self.maxDiff to None to see it.

src/icalendar/tests/test_cli_tool.py:68: AssertionError
=========================== short test summary info ============================
FAILED src/icalendar/tests/test_cli_tool.py::CLITool::test_output_is_proper
=================== 1 failed, 97 passed, 2 skipped in 12.97s ===================
ERROR: InvocationError for command /home/sabin/icalendar/.tox/pypy3/bin/coverage run --source=src/icalendar '--omit=*/tests/*' --module pytest (exited with code 1)
docs recreate: /home/sabin/icalendar/.tox/docs
docs installdeps: -r/home/sabin/icalendar/requirements_docs.txt
docs develop-inst: /home/sabin/icalendar
docs installed: alabaster==0.7.12,Babel==2.10.3,certifi==2022.6.15,charset-normalizer==2.1.1,docutils==0.17.1,-e git+ssh://git@github.com/niccokunzmann/icalendar@de54b76abb13bfac42b07050dc9fac42f514a6fa#egg=icalendar,idna==3.3,imagesize==1.4.1,Jinja2==3.1.2,MarkupSafe==2.1.1,packaging==21.3,Pygments==2.13.0,pyparsing==3.0.9,python-dateutil==2.8.2,pytz==2022.2.1,requests==2.28.1,six==1.16.0,snowballstemmer==2.2.0,Sphinx==5.1.1,sphinx-rtd-theme==1.0.0,sphinxcontrib-applehelp==1.0.2,sphinxcontrib-devhelp==1.0.2,sphinxcontrib-htmlhelp==2.0.0,sphinxcontrib-jsmath==1.0.1,sphinxcontrib-qthelp==1.0.3,sphinxcontrib-serializinghtml==1.1.5,urllib3==1.26.11
docs run-test-pre: PYTHONHASHSEED='3163265074'
docs run-test: commands[0] | make html
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v5.1.1
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 8 source files that are out of date
updating environment: 0 added, 7 changed, 0 removed
reading sources... [ 14%] api
reading sources... [ 28%] cli
reading sources... [ 42%] credits
reading sources... [ 57%] index
reading sources... [ 71%] install
reading sources... [ 85%] license
reading sources... [100%] usage

/home/sabin/icalendar/src/icalendar/prop.py:docstring of icalendar.prop:4: CRITICAL: Missing matching underline for section title overline.

###########################################################################
# This module defines these property value data types and property parameters
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [ 12%] about
writing output... [ 25%] api
writing output... [ 37%] cli
writing output... [ 50%] credits
writing output... [ 62%] index
writing output... [ 75%] install
writing output... [ 87%] license
writing output... [100%] usage

generating indices... genindex py-modindex done
highlighting module code... [ 50%] icalendar.cal
highlighting module code... [100%] icalendar.prop

writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 1 warning.

The HTML pages are in _build/html.

Build finished. The HTML pages are in docs/_build/html.
rm -rf build
___________________________________ summary ____________________________________
ERROR:   py38: commands failed
ERROR:   py39: commands failed
ERROR:   py310: commands failed
ERROR:   pypy3: commands failed
```

**Comment by jacadzaca** [comment]
hm, what's the diff when you set the self.maxDiff = None?

**Comment by niccokunzmann** [comment]
That is a good question. Can you set it in the pull request?

**Comment by jacadzaca** [comment]
it seems to be a timezone problem

**Comment by niccokunzmann** [comment]
Uh.. This is interesting!

**Comment by jacadzaca** [comment]
Europe/Warsaw

File: src/icalendar/cli.py [hunk 1]
```diff
@@ -45,65 +28,55 @@ def _format_attendees(attendees):
...
-
+    summary = event.get('summary', default='')
+    organizer = _format_name(event.get('organizer', default=''))
+    attendees = _format_attendees(event.get('attendee', default=[]))
+    location = event.get('location', default='')
+    comment = event.get('comment', '')
+    description = event.get('description', '').split('\n')
+    description = '\n'.join(map(lambda s: s.rjust(len(s) + 5), description))
+
+    timezone = datetime.utcnow().astimezone().tzinfo
```
Comment (niccokunzmann) [inline src/icalendar/cli.py]: Would this not be pytz.UTC?

File: src/icalendar/tests/test_cli_tool.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+VERSION:2.0
+CALSCALE:GREGORIAN
+BEGIN:VEVENT
+SUMMARY:Test Summary
+ORGANIZER:organizer@test.test
+ATTENDEE:attendee1@example.com
+ATTENDEE:attendee2@test.test
+COMMENT:Comment
+DTSTART;TZID=Europe/Warsaw:20220820T103400
+DTEND;TZID=Europe/Warsaw:20220820T113400
```
Comment (niccokunzmann) [inline src/icalendar/tests/test_cli_tool.py:16]: Time zone is set here.

File: src/icalendar/cli.py (hunk 1 above)
Comment (niccokunzmann) [reply src/icalendar/cli.py]: Why are your converting the time zone?

Comment (jacadzaca) [reply src/icalendar/cli.py]: My bad, sorry!

**Comment by niccokunzmann** [comment]
Nice!

**Comment by niccokunzmann** [comment]
Thanks! It was a pleasure :)

**Pull request merged**